app/forms/resend_email_confirmation_form.rb

```diff
@@ -0,0 +1,37 @@
+class ResendEmailConfirmationForm
+  include ActiveModel::Model
+  include FormEmailValidator
+
+  attr_reader :email
+
+  def initialize(email)
+    @email = email
+  end
+
+  def resend
+    'true'
+  end
+
+  def submit
+    @success = valid?
+
+    result
+  end
+
+  private
+
+  attr_reader :success
+
+  def result
+    {
+      success: success,
+      errors: errors.messages.values.flatten,
+      user_id: user.uuid,
+      confirmed: user.confirmed?
+    }
+  end
+
+  def user
+    @_user ||= User.find_with_email(email) || NonexistentUser.new
```

Comment:
I don't think this will ever be called with an empty string. The form won't allow you to submit an empty string.